stream_alert_cli/config.py

```diff
@@ -0,0 +1,180 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import json
+import os
+
+from collections import defaultdict
+
+
+class ConfigError(Exception):
+    """Exception for a non existent config file"""
+    pass
+
+
+class CLIConfig(object):
+    '''Provide an object to load, modify, and display the StreamAlertCLI Config'''
+    filename = 'variables.json'
+
+    v1_schema = {
+        'account_id',
+        'clusters',
+        'firehose_s3_bucket_suffix',
+        'flow_log_settings',
+        'kinesis_settings',
+        'kms_key_alias',
+        'lambda_function_prod_versions',
+        'lambda_handler',
+        'lambda_settings',
+        'lambda_source_bucket_name',
+        'lambda_source_current_hash',
+        'lambda_source_key',
+        'output_lambda_current_hash',
+        'output_lambda_source_key',
+        'prefix',
+        'region',
+        'tfstate_s3_key',
+        'tfvars',
+        'third_party_libs'
+    }
+
+    v2_schema = {
+        'account',
+        'alert_processor_config',
+        'alert_processor_lambda_config',
+        'clusters',
+        'firehose',
+        'flow_log_config',
+        'kinesis_streams_config',
+        'rule_processor_config',
+        'rule_processor_lambda_config',
+        'terraform'
+    }
+
+    def __init__(self):
+        self.config = self.load()
+        self.version = self.detect_version()
+        if self.version == 1:
+            self.config = self._convert_schema()
+            self.version = self.detect_version()
+
+    def __repr__(self):
+        return json.dumps(self.config)
+
+    def __getitem__(self, key):
+        return self.config[key]
+
+    def __setitem__(self, key, new_value):
+        self.config[key] = new_value
+        # TODO(jack) investigate why self.write() doesn't work as expected
+        # self.write()
+
+    def get(self, key):
+        return self.config.get(key)
+
+    def load(self):
+        """Load the variables.json configuration file
+
+        Returns:
+            [dict] loaded config from variables.json
+        """
+        if not os.path.isfile(self.filename):
+            raise ConfigError('StreamAlert variables.json file not found!')
+
+        with open(self.filename) as data:
+            try:
+                config = json.load(data)
+            except ValueError:
+                raise ConfigError('StreamAlert variables.json file is not valid JSON!')
+            return config
+
+    def write(self):
+        """Write the current config in memory to disk"""
+        with open(self.filename, 'r+') as varfile:
+            config_out = json.dumps(
+                self.config,
+                indent=4,
+                separators=(',', ': '),
+                sort_keys=True
+            )
+            varfile.seek(0)
+            varfile.write(config_out)
+            varfile.truncate()
+
+    def detect_version(self):
```

Comment:
Seems like this could be converted to a private method `_detect_version`.. unless I'm missing some use-case outside of this class